### Task1:
You need to code a program that prints numbers from 1 to 20 except multiple of 2 & 3.

Hint: use "modulo" operator and "and/or" operators

In [1]:
# Your code goes here
for a in range(1,21):
    if (a%2!=0) and (a%3!=0):
        print(a)

1
5
7
11
13
17
19


### Task2:
Write a program that keep on accepting number from the user until user enters Zero. Display the sum and average of all the numbers.

Hint: Use while loop with flag. 

In [2]:
# Your code goes here
sum = 0
flag = True
count = 0
while flag:
    a = int(input("enter your num or press '0' to end "))
    if a == 0:
        flag=False
    else:
        sum+=a
        count+=1
print(f"you input {count} nums")
print(f"your sum is {sum}")
print(f"average is {sum/count}")

enter your num or press '0' to end 2
enter your num or press '0' to end 3
enter your num or press '0' to end 4
enter your num or press '0' to end 5
enter your num or press '0' to end 0
you input 4 nums
your sum is 14
average is 3.5


# Task3:
Accept 10 numbers from the user and display their average.

Hint: Use loop for accepting numbers or use split to take input in one go

In [3]:
# Your code goes here
nums = input("enter 10 numbers use ',' to sperate ").split(",")
sum = 0
for n in nums:
    sum +=int(n)
averg = sum/10
print(f"average is {averg}.")

enter 10 numbers use ',' to sperate 1,2,3,4,5,6,7,8,9,10
average is 5.5.


# 3## Task4:
Write a program to display sum of odd numbers and even numbers that fall between 12 and 37(including both numbers)


In [4]:
# Your code goes here
even = []
odd = []
sumeven = 0
sumodd = 0
for a in range (12,38):
    if (a%2==0):
        sumeven += a
        even.append(a)
        
        
    else:
        odd.append(a)
        sumodd += a
print(f"even numbers = {even}")
print(f"odd numbers = {odd}")
print(f"sum of even numbers {sumeven}")
print(f"sum of odd numbers {sumodd}")

even numbers = [12, 14, 16, 18, 20, 22, 24, 26, 28, 30, 32, 34, 36]
odd numbers = [13, 15, 17, 19, 21, 23, 25, 27, 29, 31, 33, 35, 37]
sum of even numbers 312
sum of odd numbers 325


### Task5:
Write a program to print table of a number accepted from user.


In [5]:
# Your code goes here
table_num = int(input("enter your num "))
for num in range(1,11):
    print(f"{table_num} * {num} = {table_num *num}")        

enter your num 7
7 * 1 = 7
7 * 2 = 14
7 * 3 = 21
7 * 4 = 28
7 * 5 = 35
7 * 6 = 42
7 * 7 = 49
7 * 8 = 56
7 * 9 = 63
7 * 10 = 70


### Task6:
Write a program to display product of the digits of a number accepted from the user.

If user inputs:123 Outputs 6

If user inputs: 567 Outputs 210

In [6]:
# Your code goes here

product = 1
flag = True
while flag:
    num= int(input("enter your num or press '0' to exit "))
   
    if num== 0:
        flag = False
    else:
        product *= num
f"total product is {product}"



enter your num or press '0' to exit 3
enter your num or press '0' to exit 4
enter your num or press '0' to exit 5
enter your num or press '0' to exit 0


'total product is 60'

## Task7: 
Write a program to print first 10 even numbers in reverse order.

In [7]:
# Your code goes here
for a in range(10,0,-1):
    if a%2==0:
        print(a)

10
8
6
4
2


### Task8:
Write a program to find the sum of the digits of a number accepted from user

If user inputs:123 Outputs 6

If user inputs: 567 Outputs 18

In [9]:
# Your code goes here
sum = 0
flag = True
while flag:
    num= int(input("enter your num or press '0' to exit "))
    if num== 0:
        flag = False
    else:
        sum += num
f"total sum is {sum}"

enter your num or press '0' to exit 2
enter your num or press '0' to exit 3
enter your num or press '0' to exit 4
enter your num or press '0' to exit 0


'total sum is 9'

## Task9:Write a program that accepts a sentence and calculate the number of letters and digits.

Suppose the following input is supplied to the program:

hello world! 123

Then, the output should be:

LETTERS 10

DIGITS 3


In [10]:
# Your code goes here
import string
sentence = input("give a sentence ")
digi=lett=0
for c in sentence:
    if c.isdigit():
        digi = digi+1
    elif c.isalpha():
        lett = lett+1
        
    else:
        pass
print(f"LETTERS {lett}")
print(f"DIGITS {digi}")

give a sentence zuraizbutt72
LETTERS 10
DIGITS 2


### Task10: 
You need to trace and make a list of all such numbers from1000 to 3000 in which all the digits are even numbers.

In [11]:
# Your code goes here
even = []
for a in range(1000,3001):
    if a%2 == 0:
        even.append(a)
print(even)
        

[1000, 1002, 1004, 1006, 1008, 1010, 1012, 1014, 1016, 1018, 1020, 1022, 1024, 1026, 1028, 1030, 1032, 1034, 1036, 1038, 1040, 1042, 1044, 1046, 1048, 1050, 1052, 1054, 1056, 1058, 1060, 1062, 1064, 1066, 1068, 1070, 1072, 1074, 1076, 1078, 1080, 1082, 1084, 1086, 1088, 1090, 1092, 1094, 1096, 1098, 1100, 1102, 1104, 1106, 1108, 1110, 1112, 1114, 1116, 1118, 1120, 1122, 1124, 1126, 1128, 1130, 1132, 1134, 1136, 1138, 1140, 1142, 1144, 1146, 1148, 1150, 1152, 1154, 1156, 1158, 1160, 1162, 1164, 1166, 1168, 1170, 1172, 1174, 1176, 1178, 1180, 1182, 1184, 1186, 1188, 1190, 1192, 1194, 1196, 1198, 1200, 1202, 1204, 1206, 1208, 1210, 1212, 1214, 1216, 1218, 1220, 1222, 1224, 1226, 1228, 1230, 1232, 1234, 1236, 1238, 1240, 1242, 1244, 1246, 1248, 1250, 1252, 1254, 1256, 1258, 1260, 1262, 1264, 1266, 1268, 1270, 1272, 1274, 1276, 1278, 1280, 1282, 1284, 1286, 1288, 1290, 1292, 1294, 1296, 1298, 1300, 1302, 1304, 1306, 1308, 1310, 1312, 1314, 1316, 1318, 1320, 1322, 1324, 1326, 1328, 1330, 133

### Task11: 
You need to write a program that counts the number of vowels in a given sentence as input from console. 

Suppose the following input is supplied to the program:

Hello world! Practice makes perfect

Then, the output should be:

a appeared 2 times

e appeared 5 times

i appeared  1 time

o appeared 2 times

u appeared 0 time


In [12]:
# Your code goes here
data = input("Please type a sentence: ").lower()
vowels = "aeiou"
for v in vowels:
    print(v, data.count(v))


Please type a sentence: hello zuraiz
a 1
e 1
i 1
o 1
u 1


### Task12:
You need to write a program that takes sequence of lines as input and prints the lines after making all characters in the sentence capitalized.

Suppose the following input is supplied to the program:

Hello world

Practice makes perfect

Then, the output should be:

HELLO WORLD
PRACTICE MAKES PERFECT

In [13]:
# Your code goes here
sent = []
flag = True
while flag:
    s = input()
    if s :
         sent.append(s.upper())
        
    else:
        break 
for s in sent:
    print(s)

hello
zuraiz

HELLO
ZURAIZ


### Task13:
You need to write a code that takes a comma separated sequence of words as input and prints the words in a comma-separated sequence after sorting them alphabetically.

Suppose the following input is supplied to the program:

without,hello,bag,world

Then, the output should be:

bag,hello,without,world

In [25]:
# Your code goes here
items = input("Input comma separated sequence of words ")
words = [word for word in items.split(",")]
print(",".join(sorted(words)))

Input comma separated sequence of words hello,love,zuraiz,butt
butt,hello,love,zuraiz


### Task14:
You need code a program that calculates and prints the value according to the given formula:
P = Square root of [(2 * A * B)/C]

Following are the fixed values of A and B:

A is 50. B is 30.

The values of the literal C should be taken as console input to your program in a comma-separated sequence.

Example:
Let us assume the following comma separated input sequence is given to the program:
100,150,180

The output of the program should be:
18,22,24

In [15]:
# Your code goes here
import math

A = 50

B = 30


value = []

print("Input Values")

items = [x for x in input().split(',')]

for C in items:

    value.append(str(int(math.sqrt(2*A*float(C)/B))))


print("The followings output are")

print(','.join(value))

Input Values
100,150,180
The followings output are
18,22,24


### Task15:
You need to write a ocde in python that finds all such numbers which are divisible by 7 but are not a multiple of 5,
between 2000 and 3200 (both included).

Print the obtained numbers in a comma-separated sequence on a single line

Hint: modulo operator, and/or operator, for loop /while loop

In [28]:
# Your code goes here
nums = []
for x in range(2000,3201):
    if (x%7 == 0) and (x%5 != 0):
        nums.append(str(x))
print(",".join(nums))

2002,2009,2016,2023,2037,2044,2051,2058,2072,2079,2086,2093,2107,2114,2121,2128,2142,2149,2156,2163,2177,2184,2191,2198,2212,2219,2226,2233,2247,2254,2261,2268,2282,2289,2296,2303,2317,2324,2331,2338,2352,2359,2366,2373,2387,2394,2401,2408,2422,2429,2436,2443,2457,2464,2471,2478,2492,2499,2506,2513,2527,2534,2541,2548,2562,2569,2576,2583,2597,2604,2611,2618,2632,2639,2646,2653,2667,2674,2681,2688,2702,2709,2716,2723,2737,2744,2751,2758,2772,2779,2786,2793,2807,2814,2821,2828,2842,2849,2856,2863,2877,2884,2891,2898,2912,2919,2926,2933,2947,2954,2961,2968,2982,2989,2996,3003,3017,3024,3031,3038,3052,3059,3066,3073,3087,3094,3101,3108,3122,3129,3136,3143,3157,3164,3171,3178,3192,3199


### Task16:
With a given integral number n, write a program to generate a dictionary that contains (i, i*i) such that is an integral number between 1 and n (both included). and then the program should print the dictionary.

Suppose the following input is supplied to the program:

8

Then, the output should be:

{1: 1, 2: 4, 3: 9, 4: 16, 5: 25, 6: 36, 7: 49, 8: 64}

In [30]:
# Your code goes here
number = int(input("Type a number: "))

num = {}
for i in range(1, number+1):
    num[i] = i*i

print(num)

Type a number: 8
{1: 1, 2: 4, 3: 9, 4: 16, 5: 25, 6: 36, 7: 49, 8: 64}


### Task17:
You need to write a code which accepts a sequence of comma separated 4 digit binary numbers as its input and then check whether they are divisible by 5 or not. 

The numbers that are divisible by 5 are to be printed in a comma separated sequence.

Example:

0100,0011,1010,1001

Then the output should be:
1010

Reference: 
1) https://www.datacamp.com/.../python-data-type-conversion

2) https://stackoverflow.com/questions/21765779/converting-binary-to-decimal-integer-output?fbclid=IwAR3DD_oiq5pn99IjyxSJgojvB5W6M4JqQtv4EUl_IF_DtmuRXlP-DswcKAA

In [15]:
# Your code goes here
items = []
num = [x for x in input().split(',')]
for p in num:
    x = int(p, 2)
    if x%5 == 0:
        items.append(p)
print(','.join(items))

1000,0101,1010,1001
0101,1010
